lib/gssapi/mech/gss_cred.c

```diff
@@ -179,7 +314,8 @@ gss_import_cred(OM_uint32 * minor_status,
 	    goto out;
 	}
 
-	if (m->gm_import_cred == NULL) {
+	if (m->gm_import_cred == NULL &&
+	    (m->gm_flags & GM_USE_MG_CRED) == 0) {
```

Comment:
I mean, `m->gm_import_cred` should be `NULL` for `GM_USE_MG_CRED`, no?  And should that be `||` instead of `&&`?

Reply:
If a mech does *not* set GM_USE_MG_CRED and does *not* implement `gm_import_cred()`, then we cannot import the cred. So we should fail.

Reply:
Conversely, if it does set GM_USE_MG_CRED we will handle it in the mechglue (only for SPNEGO right now). Perhaps, perhaps, we should allow a mechanism that supports GM_USE_MG_CRED to implement `gm_import_cred()` for some future pseudo-mechanism?

Reply:
I might change the logic to special-case SPNEGO specifically.